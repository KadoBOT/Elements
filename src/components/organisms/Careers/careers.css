.careers {
  background-color: #f1f1f1;
  width: 100%;
  display: flex;
  justify-content: center;
  display: flex;
  flex-direction: column;
  align-items: center;
}

.careers__about {
  margin: 8.5vw 0px;
}
