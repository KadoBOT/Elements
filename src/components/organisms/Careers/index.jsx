import React, {PropTypes} from 'react'
import R from 'ramda'

import { About, City, TextColumns } from '../../molecules'
import './careers.css'

const Careers = ({careers}) => (
  <div id="CAREERS" className="careers">
    <div className="careers__about">
      <About
        subtitle="Join a magnificent and fun team"
        text="At Elements you will work with colleagues who are specialized in development. Not only can you share knowledge with your co-workers but also have fun. We like to have a positive approach."
      />
    </div>
    {R.map(c =>
      <City key={c.name} city={c}>
        <TextColumns city={c} />
      </City>
    , careers)}
  </div>
)

Careers.propTypes = {
  careers: PropTypes.array.isRequired
}

export default Careers
