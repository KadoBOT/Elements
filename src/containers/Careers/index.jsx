import { connect } from 'react-redux'

import Careers from '../../components/organisms/Careers'

const mapStateToProps = state => ({
  careers: state.careers
})

export default connect(mapStateToProps)(Careers)
