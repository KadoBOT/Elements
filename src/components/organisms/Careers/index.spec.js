import React from 'react'
import { shallow, render, mount } from 'enzyme';

import Careers from './'

it('should render the Careers', () => {
    const wrapper = shallow(
        <Careers careers={[]} />
    );
    expect(wrapper).toMatchSnapshot();
});
