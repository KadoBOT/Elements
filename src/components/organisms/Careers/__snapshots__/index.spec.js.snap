exports[`test should render the Careers 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="careers"
    id="CAREERS">
    <div
        className="careers__about">
        <Unknown
            subtitle="Join a magnificent and fun team"
            text="At Elements you will work with colleagues who are specialized in development. Not only can you share knowledge with your co-workers but also have fun. We like to have a positive approach." />
    </div>
</div>,
  "nodes": Array [
    <div
      className="careers"
      id="CAREERS">
      <div
            className="careers__about">
            <Unknown
                  subtitle="Join a magnificent and fun team"
                  text="At Elements you will work with colleagues who are specialized in development. Not only can you share knowledge with your co-workers but also have fun. We like to have a positive approach." />
      </div>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <Unknown
        careers={Array []} />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "careers": Array [],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="careers"
          id="CAREERS">
          <div
                    className="careers__about">
                    <Unknown
                              subtitle="Join a magnificent and fun team"
                              text="At Elements you will work with colleagues who are specialized in development. Not only can you share knowledge with your co-workers but also have fun. We like to have a positive approach." />
          </div>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="careers"
          id="CAREERS">
          <div
                    className="careers__about">
                    <Unknown
                              subtitle="Join a magnificent and fun team"
                              text="At Elements you will work with colleagues who are specialized in development. Not only can you share knowledge with your co-workers but also have fun. We like to have a positive approach." />
          </div>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Unknown
    careers={Array []} />,
}
`;
